src/cbuild/hooks/post_install/007_collect_debug_sources.py

```diff
@@ -0,0 +1,76 @@
+from cbuild.core import chroot
+
+import pathlib
+import shutil
+
+
+def _scan_file(pkg, v, processed):
+    # silently ignore invalid files
+    for source_file in (
+        chroot.enter(
+            "list-dwarf-srcfiles",
+            f"/{v.relative_to(pkg.rparent.bldroot_path)}",
+            capture_output=True,
+            ro_root=True,
+            ro_build=True,
+            unshare_all=True,
+        )
+        .stdout.strip()
+        .splitlines()
+    ):
+        source_file = pathlib.Path(source_file.strip().decode())
+
+        # don't have access to these!
+        if source_file.name in ["<stdin>"]:
+            continue
+
+        # /usr sourcefiles like /usr/include headers are already provided by -devel packages
+        if source_file.is_relative_to("/usr"):
+            continue
+        if source_file in processed:
+            # already seen
+            continue
+        processed.add(source_file)
+
+        pkg_prefix = pathlib.Path(
+            f"/usr/src/debug/{pkg.rparent.pkgname}-{pkg.rparent.pkgver}-r{pkg.rparent.pkgrel}"
+        )
+        if not source_file.is_absolute():
+            src = pkg.rparent.srcdir / source_file
+
+            if not src.exists():
+                if source_file not in [
+                    # musl crt
+                    pathlib.Path("build/../crt/Scrt1.c"),
+                    pathlib.Path("build/../crt/crt1.c"),
+                    # clang crt
+                    pathlib.Path(
+                        "build/runtimes/builtins-bins/lib/builtins/./compiler-rt/lib/builtins/crtbegin.c"
+                    ),
+                ]:
+                    pkg.log_warn(f"missing debug source file: {source_file}")
+            else:
+                dst = (
+                    pkg.rparent.destdir_base
+                    / f"{pkg.rparent.pkgname}-dbg-src-{pkg.rparent.pkgver}"
+                    / source_file
+                )
+                dst.parent.mkdir(parents=True, exist_ok=True)
+                shutil.copy(src, dst)
```

Comment:
happy path first (if src.exists():) and use .name and just the stem instead of full paths as suggested (these are not names anyone but the toolchain is using, but even if they are they will just exist and be picked up already... so it only risks silently not-warning for missing which is fine